src/main.ts

```diff
@@ -1,48 +1,31 @@
 import * as core from '@actions/core'
 import {exec} from '@actions/exec'
 import * as tc from '@actions/tool-cache'
-import * as semver from 'semver'
 import stylua from './stylua'
 
 async function run(): Promise<void> {
   try {
     const token = core.getInput('token')
-    const versionString = core.getInput('version')
-    let version = semver.clean(versionString)
-    core.debug(`Set version: "${versionString}", resolved as ${version}`)
+    const version = core.getInput('version').trim()
 
-    let releases
-    if (!version || versionString === 'latest') {
-      releases = await stylua.getReleases(token)
-      const latestVersion = stylua.getLatestVersion(releases)
-      if (!latestVersion) {
-        throw new Error(
-          'Could not find latest release version. Please specify an explicit version'
-        )
-      }
-      version = latestVersion
+    const releases = await stylua.getReleases(token)
+    const release = stylua.chooseRelease(version, releases)
+    if (!release) {
+      throw new Error(`Could not find release for version ${version}`)
     }
 
+    core.debug(`Chose release ${release.tag_name}`)
+
     // See if we already have the tool installed
-    core.debug(`Looking for cached version of binary with version ${version}`)
-    const styluaDirectory = tc.find('stylua', version ?? versionString)
+    core.debug(
+      `Looking for cached version of binary with version ${release.tag_name}`
+    )
+    const styluaDirectory = tc.find('stylua', release.tag_name)
```

Comment:
This is the only other meaningful place we might consider doing a `semver.clean()` of the release tag. AFAICT you can't tag with strange things like leading/trailing space, but you could create a release "0.17.1" and "v0.17.1". I think that in this case you /want/ the names un-cleaned, so that the cache finds the one actually associated with the release.

Reply:
Yup, sounds correct to me